### ISSUE TITLE
init() typings

### ISSUE THREAD

**Issue by webleaf** [i18next/i18next#1157]
Such code produce error:

```
i18next
    .init({
        interpolation: {
            format: (value, format, lng) => {

            }
        }
    })
    // .then(() => console.log('i18n init done'))
    // .catch(e => console.log(e))
```
```
TS2345: Argument of type '{ interpolation: { format: (value: any, format: string, lng: string) => void; }; }' is not assignable to parameter of type 'Callback'.
  Object literal may only specify known properties, and 'interpolation' does not exist in type 'Callback'.
```

**Comment by jamuhl** [comment]
@emilgpa @nikolay-borzov @schettino does this make sense? If possible i would love to get your feedback before merging?

**Comment by nikolay-borzov** [comment]
TypeScript type errors are confusing sometimes. I think the problem is within
```
format: (value, format, lng) => {}
```

format function must return `string`
```
type FormatFunction = (value: any, format?: string, lng?: string) => string;
```

Try this
```
i18next
    .init({
        interpolation: {
            format: (value, format, lng) => ''
        }
    })
    // .then(() => console.log('i18n init done'))
    // .catch(e => console.log(e))
```

**Comment by nikolay-borzov** [comment]
As for overloads order. TypeScript handbook states:

> In order for the compiler to pick the correct typecheck, it follows a similar process to the underlying JavaScript. It looks at the overload list, and proceeding with the first overload attempts to call the function with the provided parameters. If it finds a match, it picks this overload as the correct overload. For this reason, it’s customary to order overloads from most specific to least specific.

Looking at current definitions, the first one looks most specific to me
```
init(options: InitOptions, callback?: Callback): Promise<TranslationFunction>;
init(callback?: Callback): Promise<TranslationFunction>;
```

**Comment by webleaf** [comment]
@nikolay-borzov you are right. But in practice we get what we get.
Other order for that case produce more correct error for my code:
```
Error:(105, 13) TS2322: Type '(value: any, format: string, lng: string) => void' is not assignable to type 'FormatFunction'.
  Type 'void' is not assignable to type 'string'.
```
Also it would be useful to have exported interfaces. Then, for example, we could specify type `<InitOptions>` for object and "mute" ts compiler.

**Comment by webleaf** [comment]
@nikolay-borzov 

> Looking at current definitions, the first one looks most specific to me

But may be TS compiler thinks different :) More attributes may be looks like more common case (because cover more cases).

**Comment by nikolay-borzov** [comment]
I see, then I have no objections against the PR. 
You can get `InitOptions` using `i18n` interface:
```
const options: i18n.InitOptions = {};
``

**Comment by jamuhl** [comment]
@webleaf do you need this urgent - or can we publish your PR with next regular publish?

**Comment by jamuhl** [comment]
was merged and published in v13.1.1
